src/GafferVDB/PointsGridToPoints.cpp

```diff
@@ -0,0 +1,557 @@
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::V3f, openvdb::Vec3s, IECore::GeometricTypedData>( destArray, array, leafIter ); }
+		)
+	},
+	{
+		openvdb::typeNameAsString<openvdb::Vec3d>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<Imath::V3d, IECore::GeometricTypedData>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::V3d, openvdb::Vec3d, IECore::GeometricTypedData>( destArray, array, leafIter ); }
+		)
+	},
+	{
+		openvdb::typeNameAsString<std::string>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<std::string>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<std::string, std::string>( destArray, array, leafIter ); }
+		)
+	},
+	// matrix conversion - single & double
+	{
+		openvdb::typeNameAsString<openvdb::Mat4s>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<Imath::M44f>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::M44f, openvdb::Mat4s>( destArray, array, leafIter ); }
+		)
+	},
+	{
+		openvdb::typeNameAsString<openvdb::Mat4d>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<Imath::M44d>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::M44d, openvdb::Mat4d>( destArray, array, leafIter ); }
+		)
+	},
+
+	// quaternions - single & double
+	{
+		openvdb::typeNameAsString<openvdb::math::Quats>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<Imath::Quatf>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::Quatf, openvdb::math::Quats>( destArray, array, leafIter ); }
+		)
+	},
+	{
+		openvdb::typeNameAsString<openvdb::math::Quatd>(),
+		Functions(
+			[](size_t size) -> IECore::DataPtr { return createArray<Imath::Quatd>(size); },
+			[](IECore::Data *destArray, const openvdb::points::AttributeArray& array, LeafIter leafIter)  { appendData<Imath::Quatd, openvdb::math::Quatd>( destArray, array, leafIter ); }
+		)
+	},
+};
+
+void appendPrimitiveVariableData( IECoreScene::PrimitiveVariableMap &variableMap,
+	const std::string &name,
+	const std::string &type,
+	LeafIter leafIter,
+	const openvdb::points::AttributeArray &arrayData,
+	uint64_t count)
+{
+
+	auto itConverter = converters.find( type );
+	if ( itConverter == converters.end() )
+	{
+		return;
+	}
+
+	IECoreScene::PrimitiveVariable primVar;
+
+	auto primVarIt = variableMap.find( name );
+	if( primVarIt == variableMap.end() )
+	{
+		primVar = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, itConverter->second.m_create( count ) );
```

Comment:
Worth highlighting that strided attribute arrays are also possible - such as "uv" being stored as float[2]. I can provide more information if that's of interest.

Reply:
More information would be very helpful, thanks.

Reply:
There is a small introductory section on strided attributes here:

http://www.openvdb.org/documentation/doxygen/codeExamples.html#sPointStride

For strided arrays, you simply pass an additional parameter to the accessors:
```
for ( int offset = 0; offset < attributeHandle.stride(); offset++ )
{
    attributeHandle.get( *indexIter, offset );
}
```